news/2024-03-20-stdlib-migration.md

```diff
@@ -0,0 +1,47 @@
+# Upcoming migration for stdlib("c")
+
+Almost since the inception of conda-forge, the baseline version of our standard
+library ("stdlib") for C has not changed. This library comes with extra
+complications because it is an essential part of the operating system, and one
+of the few things that conda/mamba/etc. cannot ship safely.
+
+As the ecosystem has moved on and many packages are starting to require newer
+baseline versions, we need to follow suit at some point. However, to avoid
+breaking users on older systems, we need to have infrastructure in place that
+allows our packages to have sufficiently accurate metadata, such that conda can
+avoid the installation of a package requiring a newer stdlib on an old system.
+
+After many discussions across conda-forge stakeholders, the solution we arrived
+at is the introduction of a new Jinja2-function `{{ stdlib("c") }}`, which
+reflects that a given recipe requires a C stdlib. Making this relationship
+explicit will make it easy to correctly reflect the requirement for newer
+stdlib versions per feedstock, as well as in our global pinning.
+
+Up until now, the stdlib was handled implicitly as part of the compiler stack.
+In order to allow this transition to happen, we need to introduce this function
+to essentially all compiled (i.e. not pure-Python) packages. This will be done
```

Comment:
Python shouldn't receive a special callout - this equally impacts all scripting languages with compiled library interfacing (e.g., Perl, R). I think the parenthetical could be dropped altogether - "all compiled packages" is sufficiently clear. Or perhaps "(i.e., non-noarch)".

Also maybe this should be "recipes" instead of "packages.